core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/SigningStage.java

```diff
@@ -52,9 +62,53 @@ public SigningStage(HttpClientDependencies dependencies) {
     @Override
     public SdkHttpFullRequest execute(SdkHttpFullRequest request, RequestExecutionContext context) throws Exception {
         InterruptMonitor.checkInterrupted();
+        if (context.executionAttributes().getAttribute(SdkInternalExecutionAttribute.SELECTED_AUTH_SCHEME) != null) {
+            return sraSignRequest(request, context);
+        }
```

Comment:
Good working realizing that this won't always be set.

Reply:
hehe, maybe you realize more than I did. Once we get the AuthSchemeInterceptor code generated it should be non-null always, right? I did it conditional to leave the old logic in there as fallback, for potentially considering if we release these changes slowly across services and having this check in any case wouldn't hurt. But maybe you see another reason why this won't always be set?

Reply:
Old service clients + new core is a supported use-case. I hadn't thought of it either until I saw your brilliant foresight here.

We can phase this out as we phase out older protected APIs, e.g. with the smithy codegen migration where we're going to be forced to break protected APIs.

Reply:
ah, of course! to be honest, that case wasn't explicitly on my mind :D 

In either case, I have a mental note for the testing plan for SRA needs to include testing old clients with new core.